app/controllers/chores_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ChoresController < ApplicationController
+  def create
+    begin
+      #hideout_id = params[:payload]
+      params.require(%i[title description hideout_id assignee_email due_date])
+      assignee_id = User.find_by(id: params[:assignee_email])
+      chore = Chore.create(title: params[:title], description: params[:description], hideout_id: params[:hideout_id], assignee_id: params[:assignee_id], due_date: params[:due_date])
+      render status: 201,  json: chore.to_json
+    rescue ActionController::ParameterMissing => e
+      puts e
+      render status: 400
+    end
+  end
+
+  
+  def update
+    #Assume that user, chore and hideout exist
+
+    begin
+      #hideout_id = params[:payload]
+      id = params[:id] 
+      params.require(%i[title description hideout_id assignee_email due_date]) 
+      assignee_id = User.find_by(id: params[:assignee_email])
+      chore = Chore.find_by(id: id)    
+      chore.update(title: params[:title], description: params[:description], hideout_id: params[:hideout_id], assignee_id: params[:assignee_id], due_date: params[:due_date])
+      render status: 200
+    rescue
+      render status: 400
+    end
+  end
+
+def destroy
+  #Assume that user, chore and hideout exist
```

Comment:
We're only assuming that user and chore exist in all functions. Add a comment in all functions (other than create) that we're checking to make sure that user is in hideout for adequate permissions to update / delete chore.

Reply:
wouldn't it also include hideout since hideout_id is required parameter and a chore can only be assigned while in a hideout?

Reply:
Yes in theory but we're not getting the hideout_id from the frontend, we're generating it ourselves from the db in the middleware so no need to check again.